src/core/cgroup.c

```diff
@@ -778,13 +778,59 @@ void cgroup_oomd_xattr_apply(Unit *u, const char *cgroup_path) {
                 unit_remove_xattr_graceful(u, cgroup_path, "user.oomd_omit");
 }
 
+int cgroup_log_xattr_apply(Unit *u, const char *cgroup_path) {
+        static char separator[2] = { 0xff, 0x00 };
+        ExecContext *c;
+        char *pattern;
+        _cleanup_free_ char **patterns_strv = NULL;
```

Comment:
`_cleanup_strv_free_`

Reply:
We don't want to free every string from the strv. Those strings are owned by `c->log_filter_allowed_patterns` (or `c->log_filter_denied_patterns`), so we only need to free the strv itself.

Reply:
Yeah, `_cleanup_free_` is correct here. But please add a short comment about that.